### PR TITLE
enable tox test using both pyexecjs & pymongo

### PR DESCRIPTION
**Pull request by DeoLeung**

### CODE REVIEW

File: mongomock/collection.py [hunk 1]
```diff
@@ -1534,7 +1534,7 @@ def _extend_collection(out_collection, field, expression):
                     _id = stage['$group']['_id']
                     if _id:
                         key_getter = functools.partial(_parse_expression, _id)
-                        out_collection = sorted(out_collection, key=key_getter)
+                        out_collection = sorted(out_collection, key=lambda x: str(key_getter(x)))
```
Comment (vmalloc) [inline mongomock/collection.py]: This change doesn't seem related to the PR's title. Why was this introduced?

File: tests/test__mongomock.py [hunk 2]
```diff
@@ -1680,7 +1681,11 @@ def test__aggregate17(self):
         pipeline = [
             {'$project': {'_id': 0, 'created': {'$subtract': [{'$min': ['$a', '$b']}, '$count']}}}
         ]
-        self.cmp.compare.aggregate(pipeline)
+        # TODO(Metrodatateam): use skipif if version is accessible from decorator
```
Comment (vmalloc) [inline tests/test__mongomock.py]: ?

File: mongomock/collection.py (hunk 1 above)
Comment (DeoLeung) [reply mongomock/collection.py]: hmmm, that was quite an old pr I can't fully recall why i did this...looks like I was adding this to make the test pass(it was broken)

File: tests/test__mongomock.py (hunk 2 above)
Comment (DeoLeung) [reply tests/test__mongomock.py]: some aggregate was not supported until certain version, so I think we may better just skip some tests if it's not supported

Comment (vmalloc) [reply tests/test__mongomock.py]: Ok, so can we please replace this TODO with an open issue?

**Comment by pcorpet** [comment]
@DeoLeung thanks for your help. I was able to commit a similar commit in #424.
